### PR TITLE
Observe MassFlux in EvolveValenciaDivClean

### PR DESCRIPTION
**Pull request by markscheel**
## Proposed changes

Adds the interpolator to EvolveValenciaDivClean.

FishboneMoncriefDisk now observes MassFlux.

The other EvolveValenciaDivClean executables do not currently have interpolators,
but there is an easy mechanism for adding them.

### Types of changes:

- [ ] Bugfix
- [x] New feature

### Component:

- [x] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests and `clang-tidy`.
  For instructions on how to perform the CI checks locally refer to the [Dev
  guide on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

**Comment by markscheel** [comment]
Tests failed before because I hadn't rebased onto develop. I did so, and fixed what didn't compile.

**Comment by markscheel** [comment]
clang-format change only.

**Comment by markscheel** [comment]
Rebased.

I separated 3 of the commits here into separate PRs that must be merged first, before this one.

File: src/Evolution/Executables/GrMhd/ValenciaDivClean/CMakeLists.txt [hunk 1]
```diff
@@ -2,6 +2,7 @@
 # See LICENSE.txt for details.
 
 set(LIBS_TO_LINK
+  ApparentHorizons
```
Comment (nilsdeppe) [inline src/Evolution/Executables/GrMhd/ValenciaDivClean/CMakeLists.txt]: These aren't needed for all the executables

File: src/Evolution/Executables/GrMhd/ValenciaDivClean/EvolveValenciaDivClean.hpp [hunk 2]
```diff
@@ -98,7 +115,31 @@ class CProxy_ConstGlobalCache;
...
+          hydro::Tags::MassFlux<DataVector, 3, ::Frame::Inertial>,
+          ::Frame::Inertial>>;
+  using compute_items_on_source = tmpl::list<>;
+  using vars_to_interpolate_to_target = tmpl::list<
+      hydro::Tags::RestMassDensity<DataVector>,
+    hydro::Tags::SpatialVelocity<DataVector, 3, Frame::Inertial>,
+      hydro::Tags::LorentzFactor<DataVector>, gr::Tags::Lapse<DataVector>,
+      gr::Tags::Shift<3, Frame::Inertial, DataVector>,
+      gr::Tags::SqrtDetSpatialMetric<DataVector>>;
+  using compute_items_on_target = tmpl::append<
```
Comment (kidder) [inline src/Evolution/Executables/GrMhd/ValenciaDivClean/EvolveValenciaDivClean.hpp]: tmpl::push_front<list,A,B>

**Review by kidder — CHANGES_REQUESTED** [review]

**Comment by kidder** [comment]
I am happy for you to squash if @nilsdeppe is

**Review by nilsdeppe — COMMENTED** [review]
I'm happy for you to squash

File: src/Evolution/Executables/GrMhd/ValenciaDivClean/CMakeLists.txt (hunk 1 above)
Comment (markscheel) [reply src/Evolution/Executables/GrMhd/ValenciaDivClean/CMakeLists.txt]: So I should just remove them?  Does cmake link them through another mechanism?

**Comment by markscheel** [comment]
Squashed, and added another commit changing AddSpectreExecutable.cmake

**Review by nilsdeppe — APPROVED** [review]
image

**Review by kidder — APPROVED** [review]

**Pull request merged**